app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,44 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    @doc_html = download_content
+    process
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout
+    page.update_attributes(status: Page::ERROR_STATUS)
+  end
+
+  private
+
+  def process
+    page.update_attributes(status: Page::IN_PROGRESS_STATUS)
+    parse_html_content
+    make_screenshot
+    page.update_attributes(status: Page::PROCESSED_STATUS)
+  end
+
+  def download_screenshot(file)
+    File.new(Gastly.capture(page.source_url, file.path))
+  end
+
+  def make_screenshot
+    file = Tempfile.new
+    page.update_attributes(screenshot: download_screenshot(file))
+    file.close!
+  end
+
+  def download_content
+    file = open(page.source_url)
```

Comment:
file variable is not needed

Reply:
what if status is 500 or 404? success only if 20X status?

Reply:
errors handle in perform method, end changing page status to 'error'

Reply:
500 and 404 pages (status codes) are not errors (or crashes)
but such pages doesn't have good content

Reply:
If I try to open non-existent page I will get HTTPError
>> open('http://en.wikipedia.org/wiki/ewewew')
OpenURI::HTTPError: 404 Not Found
        from (irb):2
and I handle this type of error
rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout